src/Sarif.Converters/SemmleQlConverter.cs

```diff
@@ -82,7 +79,18 @@ public override void Convert(Stream input, IResultLogWriter output, OptionallyEm
 
             output.Initialize(run);
 
-			output.WriteFiles(fileDictionary);
+            var visitor = new AddFileReferencesVisitor();
+            visitor.VisitRun(run);
+
+            foreach (Result result in results)
+            {
+                visitor.VisitResult(result);
+            }
+
+            if (run.Files != null && run.Files.Count > 0)
```

Comment:
>run.Files != null && run.Files.Count > 0 [](start = 16, length = 40)

`if (run.Files?.Count > 0)` or `if (run.Files?.Any())` (my favorite expression for this)...

... but the base class already does this.

Reply:
run.Files.Any() works because the extension handles the null 'this' case. still, this is unnecessary as you've pointed out, we should be calling the base helper. Simply failed to DRY here.

---
In reply to: [244521298](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244521298) [](ancestors = 244521298)